### ISSUE TITLE
Unable to install nightly

### ISSUE THREAD

**Issue by FallingSnow** [lukechilds/zsh-nvm#43]
```
$ nvm install nightly         
Version 'node' not found - try `nvm ls-remote` to browse available versions.
Clearing mirror cache...
Done!
```
Let me know if you need any more information.

**Comment by lukechilds** [comment]
Thanks for reporting.

I'm not quite sure why this has broken, possibly the format of https://nodejs.org/download/nightly/ has changed, although that seems unlikely.

Or possibly `nvm` has changed the way it parses versions, as it seems to be interpreting the nightly versions as `iojs`.

```
$ NVM_NODEJS_ORG_MIRROR=https://nodejs.org/download/nightly/ nvm install node
Version 'node' not found - try `nvm ls-remote` to browse available versions.

$ NVM_NODEJS_ORG_MIRROR=https://nodejs.org/download/nightly/ nvm ls-remote
    iojs-v1.0.0
    iojs-v1.0.1
    iojs-v1.0.2
    iojs-v1.0.3
    iojs-v1.0.4
    iojs-v1.1.0
    iojs-v1.2.0
    iojs-v1.3.0
    iojs-v1.4.1
    ...
    iojs-v3.3.1
```

**Comment by FallingSnow** [comment]
I was able to install nightly via v8-canary: `NVM_NODEJS_ORG_MIRROR=https://nodejs.org/download/v8-canary nvm install node`.

**Comment by lukechilds** [comment]
Hmmnn, seems to just be that by chance it's parsing a couple of the v8-canary options as `node` instead of `iojs`:

```
NVM_NODEJS_ORG_MIRROR=https://nodejs.org/download/v8-canary/ nvm ls-remote
v9.0.0-v8-canary201710301f8f3dcf36
v10.0.0-v8-canary20180321f7efae222a
    iojs-v1.0.0
    iojs-v1.0.1
    iojs-v1.0.2
    ...
    iojs-v3.3.1
```

**Comment by lukechilds** [comment]
Unfortunately I don't really have the time to look into this further right now. PRs welcome!

I think the best solution would either be to remove this feature completely (it was always a bit of a hack), or add our own version parsing that properly detects `canary`/`nightly`/`rc` versions.

The latter would need to be done very carefully to avoid any breakage and not mess things up with `nvm`. It should probably be submitted as a PR to `nvm` rather than to `zsh-nvm` if the maintainers want it.

**Comment by FallingSnow** [comment]
I agree. I believe this is a feature that should be in nvm, not zsh-nvm. Unfortunately I doesn't seem like a priority to them.

**Comment by cswl** [comment]
@lukechilds 
I stopped messing with nightly node... so I didnt notice this yet...

I can write our own nightly/rc installer which parses the stuff..
which uses absolute versions and doesnt interfere with stable `nvm` installs...

I will get around to writing this tonight..

**Comment by lukechilds** [comment]
That would be amazing, I think it's a fairly big job though.

If you're serious about doing this it might be worth contributing it upstream, directly to `nvm`. Then everyone can benefit, not just users of `zsh-nvm`. It would probably be worth creating an issue there first and letting them know what you're planning, just to make sure they definitely want it.
